### DISCUSSION TITLE
Prevent objects for being scaled during zoom

### DISCUSSION THREAD

**Discussion by umtdemr** [fabricjs/fabric.js#8260] [Q&A] [answered]
I've created grouped object that includes image, circle and text and I want to prevent it for being scaled while I zooming-in or zooming-out in canvas. I've found a solution for it but it doesn't seem right solution for me and it has side effects. For example selecting object ruins scaling a bit, I'm not sure with this but It also seems changing scaling properties changes position of object.

My solution is:
```js
fabric.Object.prototype.ignoreZoom = false;   // for the object that I don't want it to be scaled
fabric.Object.prototype.transform = function(ctx) {
  var needFullTransform = (this.group && !this.group._transformDone) ||
    (this.group && this.canvas && ctx === this.canvas.contextTop);
  var m = this.calcTransformMatrix(!needFullTransform);
  ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
  
  if (this.ignoreZoom) {
    // for preventing object for being scaled I just change scale properties according to zoom level.
    const zoom = canvas.getZoom();
    this.scaleX = 1 / zoom;
    this.scaleY = 1 / zoom;
  }
}
```

Is there any better way for this?

**Comment by ShaMan123** [comment]
If the objects are grouped I would do the manipulation on the group itself.
If you set originX/Y to center it should fix positioning (or `setPositionByOrigin`).
Or you could use instead `this.scale()`.

**Reply by umtdemr** [reply]
I also manipulate the group itself. Setting originX to center and originY to bottom is actually sovled my problem. Thank you for your reply.

**Comment by umtdemr** [comment, marked as answer]
I think there is no better way than this. Alternatively, you can change the transform method for the object that you want to prevent being zoomed (I'm not sure this is recommended). 

Here is my solution:

 ```js
 fabric.Object.prototype.ignoreZoom = false;
fabric.Object.prototype.transform = function(ctx) {
  if (this.ignoreZoom) {
    const zoom = canvas.getZoom();   // or this.canvas.getZoom()
    this.scaleX = 1 / zoom;
    this.scaleY = 1 / zoom;
  }

// below lines are the original code. This may be changed in the future. Check this: https://github.com/fabricjs/fabric.js/blob/master/src/shapes/object.class.ts#L843
  var needFullTransform = (this.group && !this.group._transformDone) ||
    (this.group && this.canvas && ctx === this.canvas.contextTop);
  var m = this.calcTransformMatrix(!needFullTransform);
  ctx.transform(m[0], m[1], m[2], m[3], m[4], m[5]);
}
```
